test/test_llogcolor.py

```diff
@@ -0,0 +1,101 @@
+"""A test suite for comparing the text output
+of llogcolor for an assumed-to-be-correct output.
+
+The pager is skipped for these tests, and the
+color control codes go strait to a file, so they get saved.
+"""
+
+import filecmp
+import pathlib
+import sys
+import tempfile
+import unittest
+
+# now I need to import llogcolor
+# this isn't the normal way to import,
+# but there are 2 issues.
+# 1. this is not a python only project, I don't want
+# to structure all of lustre-tools-llnl as a python package
+# 2. the file to be tested doesn't have '.py' suffix
+# from https://stackoverflow.com/questions/2601047/
+from importlib.util import spec_from_loader, module_from_spec
+from importlib.machinery import SourceFileLoader
+
+path_to_llogcolor = str(pathlib.Path("../scripts/llogcolor").resolve())
+spec = spec_from_loader(
+    "llogcolor", SourceFileLoader("llogcolor", path_to_llogcolor)
+)
+llogcolor = module_from_spec(spec)
+spec.loader.exec_module(llogcolor)
+
+#### paths to input logs ###
+# small and simple log file, all of the same thread ID
+in_simple = "llogcolor_files/sample_logs"
```

Comment:
Instead of having a few global variables (in_simple, out_P_C_simple, etc.) which are tightly linked semantically, would it make sense to define one or more tuples, in the form of (input_name, (args tuple), output_name)?

Reply:
Yes. However taken too far I'm not sure how this will work with `unittest`. The way it works now is I just do
`python test_llogcolor.py -v` and each test runs, prints the test function doc string, and the results. I realize that the two tests have the same structure but different arguments and doc strings, and so would many additional tests if I were to write them.
Where I think you're going is to basically have the same function run on a bunch of sets of arguments, instead of writing the same function over and over again but using different args and doc strings. Maybe I'm projecting here, but auto-generating a function for each data item, and still getting the benefits of seeing the results for each function looks like a good idea. I'll look into how that can be done.

Reply:
Before you do that, let's discuss the greater "import python vs. create a shell and run it" question.

Reply:
Can you elaborate on that question.